src/ui/views/EditAwards.football.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { ChangeEvent, FormEvent, useState } from "react";
+
+import useTitleBar from "../hooks/useTitleBar";
+import type { Player, View } from "../../common/types";
+import { logEvent, toWorker, helpers, realtimeUpdate } from "../util";
+import SelectReact from "../components/SelectMultiple";
+const EditAwardsFootball = ({
+	godMode,
+	players,
+	awards,
+	season,
+}: View<"editAwards">) => {
+	useTitleBar({
+		title: "Edit awards",
+		dropdownView: "edit_awards",
+		dropdownFields: { seasons: season },
+	});
+
+	const awardsInitial = awards;
+	const [state, setState] = useState(() => {
+		const aws = helpers.deepCopy(awards);
+
+		return {
+			aws,
+		};
+	});
+	const handleChange = (obj: any, pl: any) => {
+		let error: boolean = false;
+		const p: any = pl;
+		const type = obj.props.award;
+		const numberTeam = obj.props.teamNumber;
+		const numberPlayer = obj.props.playerNumber;
+		if (state.aws == undefined) {
```

Comment:
`state.aws` should never be undefined, unless the `awards` prop is undefined, in which case `handleChange` should never be called. So I *think* this check is not needed.

Reply:
This wasn´t in the code originally. I encounter a problem that when starting in the season 2021 with no awards for example, and changing to 2020 the useState in line 20 didn´t execute so state.aws would still be undefined, so this is how I fixed

Reply:
> changing to 2020 the useState in line 20 didn´t execute

That is a good point, but that should be addressed more directly, like:

```
useEffect(() => {
    setState(state => ({
        ...state,
        aws: awards,
    }));
}, [awards, season]);
```

That useEffect block will run any time `awards` or `season` changes, which is probably the correct behavior. Otherwise wouldn't your state get stale any time the season changes, even if `state.aws` is not undefined?

Reply:
Yes, this is much better and make more sense, I wish I knew about this function.